File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;
/*
원소는 0~9
연속적으로 나타나는 숫자는 하나 제외 전부 제거
남은 수 반환할 때 순서는 유지해야된다.
데이터를 for문으로 순회하자 이 때 이전에 값이 있으면 제외해라
혹은 deque사용해도 될듯?

*/
public class Solution {
    public int[] solution(int []arr) {
        Deque<Integer> q = new LinkedList<>(); 
        q.add(arr[0]);
        
        for(int i = 1 ; i < arr.length; i++){
            if(q.peekLast() != arr[i]){
                q.add(arr[i]);
            }
        }
        int[] answer = new int[q.size()];
        for(int i = 0 ; i < answer.length; i++){
            answer[i] = q.poll();
        }

        return answer;
    }
}

File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
'''
    0~9 로 구성
    연속하는 숫자 하나 남기고 전부 제거 -> 제거된 수 반환할 때는 arr의 원소 순서 유지해야된다.
    즉 answer 리스트를 만들어서 거기다가 데이터를 하나하나 다 박아라
    검증할 때는 일단 스택을 하나 만들어서 peek가 같은지 체크해보고 같으면 꺼내라 이런 느김
'''
def solution(arr):
    answer = []
    for i in range(len(arr)):
        if i == 0:
            answer.append(arr[i])
            
        elif arr[i] != arr[i - 1]:
            answer.append(arr[i])
    return answer